metrics/options.go

```diff
@@ -58,8 +69,35 @@ func defaultOptions() *options {
 		errorHandler: func(err error) {
 			logger.WithError(err).Error(err.Error())
 		},
-		metricSampleRate: defaultMetricSampleRate,
+		sampleRate: defaultMetricSampleRate,
+	}
+}
+
+// parseMetricOptions processes the provided metric options and returns a configured metricOptions object.
+// This function combines globally defined defaults with any user-provided option overrides.
+// Notes:
+//   - If no options are provided, the returned metricOptions will use:
+//   - No tag
+//   - Global sample rate (opts.sampleRate) as the default sampling rate
+//   - Any provided options will override these defaults for the specific metric
+func parseMetricOptions(options ...MetricOptions) metricOptions {
+	result := metricOptions{
+		tags:       []string{},
+		sampleRate: opts.sampleRate, // Use global sample as default
+	}
+
+	for _, opt := range options {
+		err := opt(&result)
+		if err != nil {
+			return metricOptions{
+				errorHandler: func(err error) {
+					logger.WithError(err).Error(err.Error())
+				},
+			}
+		}
```

Comment:
I think there might be an issue here.

If `opt(&result)` returns an error this will return metrics options with the error handler set. What I think should happen is that if `opt(&result)` returns an error then `opts.errorHandler(err)`, where `opts` are the global options should be called. Next we need to decide on how to proceed, if the `opts.errorHandler` has a function that calls `panic(err)` then the application will at some point terminate unless the panic is handled. On the other hand if `opts.errorHandler` keeps the application in a good state `parseMetricOptions` must return something. We also need to decide if we process all `options` in this case.

```suggestion
		err := opt(&result)
		if err != nil {
			opts.errorHandler(err)
		}
```

Reply:
Another alternative is to have `parseMetricOptions` return an error, similar to `resolveOptions` and call `opts.errorHandler` in each of the metrics functions (`Gauge`, `Count` etc).

Reply:
I agree with Niels Henrik, that we should use the global `opts.errorHandler`, and not let every metric-type define their own error-handler.